core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -1124,15 +1124,13 @@ class FsHistoryProviderSuite extends SparkFunSuite with Matchers with Logging {
       SparkListenerApplicationEnd(5L))
     val mockedFs = spy(provider.fs)
     doThrow(new AccessControlException("Cannot read accessDenied file")).when(mockedFs).open(
-      argThat((path: Path) => path.getName.toLowerCase(Locale.ROOT) == "accessdenied"))
+      argThat((path: Path) => path.getName.toLowerCase(Locale.ROOT) == "accessdenied" &&
+        clock.getTimeMillis() < 1533132471 + 24 * 60 * 60 * 1000))
     val mockedProvider = spy(provider)
     when(mockedProvider.fs).thenReturn(mockedFs)
     updateAndCheck(mockedProvider) { list =>
       list.size should be(1)
     }
-    writeFile(accessDenied, true, None,
-      SparkListenerApplicationStart("accessDenied", Some("accessDenied"), 1L, "test", None),
-      SparkListenerApplicationEnd(5L))
```

Comment:
This has a wrong assumption that the file size will grow. In this PR, I removed this assumption and add time condition at line 1128.

Reply:
sorry, I don't understand this, may you please elaborate a bit more? Thanks!

Reply:
@mgaido91 . No problem.
What I mean is that `accessDenied` file is created twice at line 1119 and line 1133 with the same name and with different size. So, this test case didn't detect this bug so far.

Reply:
yes, I see now. Well the increased size was intentional and enforced by having multiple lines... But after your fix, this is not necessary anymore, I see, thanks.